pkg/api/rest/rest.go

```diff
@@ -125,6 +125,13 @@ func (w GracefulDeleteAdapter) Delete(ctx api.Context, name string, options *api
 	return w.Deleter.Delete(ctx, name)
 }
 
+// CollectionDeleter is an object that can delete a collection
+// of RESTful resources.
+type CollectionDeleter interface {
```

Comment:
I don't think so. I think the interface should remain as it is here.

What we would like to cleanup is the implementation of DeleteCollection in pkg/registry - but there is already a TODO there.